qcodes/instrument/parameter.py

```diff
@@ -2766,3 +2766,29 @@ def expand_setpoints_helper(parameter: ParameterWithSetpoints,
         data = results
     res.append((parameter, data))
     return res
+
+
+class AbstractParameter(Parameter):
```

Comment:
does it have to inherit from Parameter? this feature will be more useful if it inherits from _BaseParameter becasue then abstract definitions of ParameterWithSetpoints and MultiParameter may be supported

Reply:
Agreed perhaps we should consider making AbstractParameter the baseclass for parameter

Reply:
_BaseParameter already implements `get_raw` and `set_raw` that will raise a NotIMplementedError if called. 

I think it would be better if we simply rename `_BaseParameter` to `AbstractParameter` (perhaps retaining `_BaseParameter` as an alias for now and use that everywhere.

Reply:
Oh these are great suggestions! I will modify my PR accordingly